internal/db/paginate.go

```
@@ -1,22 +1,39 @@
-	totalPages := int(math.Ceil(float64(totalRows) / float64(pagination.GetLimit())))
-	pagination.TotalPages = totalPages
-	pagination.TotalResults = int(totalRows)
+		// For pagination we need to build a SELECT that includes both:
+		// 1. All columns from the main table (e.g., beers.*)
+		// 2. A COUNT(*) OVER() window function for total row count
+
+		// Try to determine the table name from GORM's statement
+		var tableName string
+		stmt := &gorm.Statement{DB: db}
+		// Parse the model that was passed to Model() earlier (e.g., &model.Beer{})
+		stmt.Parse(db.Statement.Model)
```

golangci-lint (2.2.2)

Error return value of `stmt.Parse` is not checked

(errcheck)